promql/engine_test.go

```diff
@@ -750,6 +827,34 @@ load 10s
 			End:      time.Unix(10, 0),
 			Interval: 5 * time.Second,
 		},
+		{
+			Query:        "irate(bigmetric[1s])",
+			MaxSamples:   1,
+			TotalSamples: 4,
+			PeakSamples:  1,
+			Result: Result{
+				nil,
+				Matrix{},
+				nil,
+			},
+			Start:    time.Unix(0, 0),
+			End:      time.Unix(10, 0),
+			Interval: 5 * time.Second,
+		},
+		{
+			Query:        `metric + metric`,
+			MaxSamples:   11,
+			TotalSamples: 15,
```

Comment:
I'd expect this to be 9

Reply:
Same culprit here too. Let's have the increment inside the `if point.T == ts` block in rangeEval as it was earlier. That gives the exact calculation.